popdat/src/activities.rs

```diff
@@ -1,11 +1,94 @@
 
 impl CensusPerson {
-    pub fn generate_schedule(&self, config: &Config, rng: &mut XorShiftRng) -> Schedule {
-        // TODO Maybe first classify into a PersonType, then generate a Schedule? I don't know if
-        // the intermediate is useful or not
-        todo!()
+    pub fn generate_schedule(&self, _config: &Config, rng: &mut XorShiftRng) -> Schedule {
+        // TODO How do we pick these categories based on census data?
+        let person_type = if rng.gen_bool(0.5) {
+            PersonType::Student
+        } else {
+            PersonType::Worker
+        };
+
+        // Fill out a list of activities and how long the person should do the activity before
+        // travelling to the next place.
+        let mut plan = Vec::new();
+        let start_time;
+
+        match person_type {
+            PersonType::Student => {
+                // I'm probably channeling a college student here...
+                start_time = rand_time(rng, hours(8), hours(11));
+                if rng.gen_bool(0.95) {
+                    plan.push((Activity::Breakfast, minutes(30)));
+                }
+                plan.push((Activity::School, rand_duration(rng, hours(3), hours(6))));
+                if rng.gen_bool(0.3) {
+                    plan.push((
+                        Activity::Lunch,
+                        rand_duration(rng, minutes(20), minutes(40)),
+                    ));
+                }
+                plan.push((Activity::School, rand_duration(rng, hours(2), hours(4))));
+                if rng.gen_bool(0.6) {
+                    plan.push((Activity::Entertainment, hours(2)));
+                } else {
+                    plan.push((Activity::Errands, rand_duration(rng, minutes(15), hours(1))));
+                }
+                // The last duration doesn't matter
+                plan.push((Activity::Home, hours(8)));
+            }
+            PersonType::Worker => {
+                start_time = rand_time(rng, hours(6), hours(9));
+                if rng.gen_bool(0.8) {
+                    plan.push((Activity::Breakfast, minutes(15)));
+                }
+                plan.push((Activity::Work, rand_duration(rng, hours(4), hours(5))));
+                plan.push((
+                    Activity::Lunch,
+                    rand_duration(rng, minutes(20), minutes(40)),
+                ));
+                plan.push((Activity::Work, hours(4)));
+                if rng.gen_bool(0.8) {
+                    plan.push((Activity::Errands, rand_duration(rng, minutes(15), hours(1))));
+                }
+                // The last duration doesn't matter
+                plan.push((Activity::Home, hours(8)));
+            }
+        }
+
+        let mut schedule = Vec::new();
+        let mut now = start_time;
+        for (activity, duration) in plan {
+            schedule.push((now, activity));
+            // TODO We have to add in commute time here, but at this stage in the pipeline, we have
```

Comment:
So this probably means the pipeline is wrong and we need to plumb forward the activity durations and guess the commute time later